### ISSUE TITLE
Samples from Z-Anime training look messed up.

### ISSUE THREAD

**Issue by a-l-e-x-d-s-9** [ostris/ai-toolkit#879]
Using the model: https://huggingface.co/SeeSee21/Z-Anime/tree/main/diffusers
Downloaded it to my disk and giving it as path.
All samples look like that:

<img width="845" height="868" alt="Image" src="https://github.com/user-attachments/assets/4f7472ac-71b9-43e2-b95d-590b1a55415f" />

When I trained Z-Image Turbo and Z-Image Base they worked fine with samples.
I wonder if this model needs a different sampler, I tried DDPM, but it made similar messed up images.

``` YAML
  process:
    - type: "diffusion_trainer"
      training_folder: "/home/alexds9/stable-diffusion-webui/models/Lora/My/Training"
      sqlite_db_path: "./aitk_db.db"
      device: "cuda"
      trigger_word: null
      performance_log_every: 10
      network:
        type: "lora"
        linear: 16
        linear_alpha: 16
        conv: 16
        conv_alpha: 16
        lokr_full_rank: true
        lokr_factor: 16
        network_kwargs:
          ignore_if_contains: []
      save:
        dtype: "bf16"
        save_every: 100
        max_step_saves_to_keep: 50
        save_format: "diffusers"
        push_to_hub: false
      datasets:
        - folder_path: "/home/alexds9/Documents/stable_diffusion/root/training_202605_12_z-anime-base_g/dataset/"
          mask_path: null
          mask_min_value: 0.1
          default_caption: ""
          caption_ext: "txt"
          caption_dropout_rate: 0.05
          cache_latents_to_disk: true
          is_reg: false
          network_weight: 1
          resolution:
            - 512
            - 768
          controls: []
          shrink_video_to_frames: true
          num_frames: 1
          do_i2v: true
          flip_x: false
          flip_y: false
      train:
        batch_size: 2
        bypass_guidance_embedding: false
        steps: 2800
        gradient_accumulation: 1
        train_unet: true
        train_text_encoder: false
        gradient_checkpointing: true
        noise_scheduler: "flowmatch"
        optimizer: "adamw8bit"
        timestep_type: "weighted"
        content_or_style: "balanced"
        optimizer_params:
          weight_decay: 0.0001
        unload_text_encoder: false
        cache_text_embeddings: true
        lr: 0.000275
        ema_config:
          use_ema: false
          ema_decay: 0.99
        skip_first_sample: false
        force_first_sample: false
        disable_sampling: false
        dtype: "bf16"
        diff_output_preservation: false
        diff_output_preservation_multiplier: 1
        diff_output_preservation_class: "person"
        switch_boundary_every: 1
        loss_type: "mae"
      model:
        name_or_path: "/home/alexds9/Documents/stable_diffusion/ai-toolkit/models/Z-Anime-Base-diffusers/"
        quantize: false
        qtype: "qfloat8"
        quantize_te: false
        qtype_te: "qfloat8"
        arch: "zimage"
        low_vram: false
        model_kwargs: {}
        layer_offloading: true
        layer_offloading_text_encoder_percent: 0.3
        layer_offloading_transformer_percent: 0.3
      sample:
        sampler: "flowmatch"
        sample_every: 100
        width: 512
        height: 768
        samples:
          - prompt: "woman realistic"
          - prompt: "woman illustrated"
          - prompt: "woman in the park"
        neg: "bad quality, poor quality, disfigured, jpg, toy, bad anatomy, missing limbs, missing fingers, ugly, scary, watermark, lowres, bad hands, text, error, blurry"
        seed: 10
        walk_seed: true
        guidance_scale: 7
        sample_steps: 25
        num_frames: 1
```

**Comment by ziyaxuanyi** [comment]
> Using the model: https://huggingface.co/SeeSee21/Z-Anime/tree/main/diffusers Downloaded it to my disk and giving it as path. All samples look like that:
> 
> <img alt="Image" width="845" height="868" src="https://private-user-images.githubusercontent.com/6518064/607075391-4f7472ac-71b9-43e2-b95d-590b1a55415f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODEzMzAxOTAsIm5iZiI6MTc4MTMyOTg5MCwicGF0aCI6Ii82NTE4MDY0LzYwNzA3NTM5MS00Zjc0NzJhYy03MWI5LTQzZTItYjk1ZC01OTBiMWE1NTQxNWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDYxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjA2MTNUMDU1MTMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Nzk2MWExYTQzODczZjM2ZGQ2ZDIwYTllZjk5ZWY4MzExNmM0YTdhOWQyM2FhM2VmODczZmNlYjkyM2IwMzE4NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmcmVzcG9uc2UtY29udGVudC10eXBlPWltYWdlJTJGcG5nIn0.aJEuBJSz-MybnzC3EBw26AOcPem3ogxXe-oti3xqwZM">
> When I trained Z-Image Turbo and Z-Image Base they worked fine with samples. I wonder if this model needs a different sampler, I tried DDPM, but it made similar messed up images.
> 
>   process:
>     - type: "diffusion_trainer"
>       training_folder: "/home/alexds9/stable-diffusion-webui/models/Lora/My/Training"
>       sqlite_db_path: "./aitk_db.db"
>       device: "cuda"
>       trigger_word: null
>       performance_log_every: 10
>       network:
>         type: "lora"
>         linear: 16
>         linear_alpha: 16
>         conv: 16
>         conv_alpha: 16
>         lokr_full_rank: true
>         lokr_factor: 16
>         network_kwargs:
>           ignore_if_contains: []
>       save:
>         dtype: "bf16"
>         save_every: 100
>         max_step_saves_to_keep: 50
>         save_format: "diffusers"
>         push_to_hub: false
>       datasets:
>         - folder_path: "/home/alexds9/Documents/stable_diffusion/root/training_202605_12_z-anime-base_g/dataset/"
>           mask_path: null
>           mask_min_value: 0.1
>           default_caption: ""
>           caption_ext: "txt"
>           caption_dropout_rate: 0.05
>           cache_latents_to_disk: true
>           is_reg: false
>           network_weight: 1
>           resolution:
>             - 512
>             - 768
>           controls: []
>           shrink_video_to_frames: true
>           num_frames: 1
>           do_i2v: true
>           flip_x: false
>           flip_y: false
>       train:
>         batch_size: 2
>         bypass_guidance_embedding: false
>         steps: 2800
>         gradient_accumulation: 1
>         train_unet: true
>         train_text_encoder: false
>         gradient_checkpointing: true
>         noise_scheduler: "flowmatch"
>         optimizer: "adamw8bit"
>         timestep_type: "weighted"
>         content_or_style: "balanced"
>         optimizer_params:
>           weight_decay: 0.0001
>         unload_text_encoder: false
>         cache_text_embeddings: true
>         lr: 0.000275
>         ema_config:
>           use_ema: false
>           ema_decay: 0.99
>         skip_first_sample: false
>         force_first_sample: false
>         disable_sampling: false
>         dtype: "bf16"
>         diff_output_preservation: false
>         diff_output_preservation_multiplier: 1
>         diff_output_preservation_class: "person"
>         switch_boundary_every: 1
>         loss_type: "mae"
>       model:
>         name_or_path: "/home/alexds9/Documents/stable_diffusion/ai-toolkit/models/Z-Anime-Base-diffusers/"
>         quantize: false
>         qtype: "qfloat8"
>         quantize_te: false
>         qtype_te: "qfloat8"
>         arch: "zimage"
>         low_vram: false
>         model_kwargs: {}
>         layer_offloading: true
>         layer_offloading_text_encoder_percent: 0.3
>         layer_offloading_transformer_percent: 0.3
>       sample:
>         sampler: "flowmatch"
>         sample_every: 100
>         width: 512
>         height: 768
>         samples:
>           - prompt: "woman realistic"
>           - prompt: "woman illustrated"
>           - prompt: "woman in the park"
>         neg: "bad quality, poor quality, disfigured, jpg, toy, bad anatomy, missing limbs, missing fingers, ugly, scary, watermark, lowres, bad hands, text, error, blurry"
>         seed: 10
>         walk_seed: true
>         guidance_scale: 7
>         sample_steps: 25
>         num_frames: 1

You just need to replace the scheduler and transformer folders of z-image-base with those from Z-Anime, keeping all other files the same as z-image-base.

Because there are differences in how Anime and base store text_encoder, ai-toolkit cannot correctly read Anime's text_encoder, so please keep it the same as base.

By the way, I’ve already tried training Z-Anime’s lora, and the results were just as bad as base, still difficult to fine-tune like base.

**Comment by a-l-e-x-d-s-9** [comment]
@ziyaxuanyi 
Thank you, it fixed it for me!
